stores/hostdb.go

```diff
@@ -426,6 +529,179 @@ func (ss *SQLStore) Host(ctx context.Context, hostKey types.PublicKey) (hostdb.H
 	}, nil
 }
 
+func (ss *SQLStore) HostInfo(ctx context.Context, autopilotID string, hk types.PublicKey) (hi api.HostInfo, err error) {
+	err = ss.db.Transaction(func(tx *gorm.DB) error {
+		// fetch ap id
+		var apID uint
+		if err := tx.
+			Model(&dbAutopilot{}).
+			Where("identifier = ?", autopilotID).
+			Select("id").
+			Take(&apID).
+			Error; errors.Is(err, gorm.ErrRecordNotFound) {
+			return api.ErrAutopilotNotFound
+		} else if err != nil {
+			return err
+		}
+
+		// fetch host id
+		var hID uint
+		if err := tx.
+			Model(&dbHost{}).
+			Where("public_key = ?", publicKey(hk)).
+			Select("id").
+			Take(&hID).
+			Error; errors.Is(err, gorm.ErrRecordNotFound) {
+			return api.ErrHostNotFound
+		} else if err != nil {
+			return err
+		}
+
+		// fetch host info
+		var entity dbHostInfo
+		if err := tx.
+			Model(&dbHostInfo{}).
+			Where("db_autopilot_id = ? AND db_host_id = ?", apID, hID).
```

Comment:
I'd turn this into a single query to avoid the roundtrips.

```go
if err := tx.
  Model(...).
  Where("db_autopilot_id", gorm.Expr("SELECT id FROM autopilots WHERE identifier = ?", autopilotID)).
  Where("db_host_id", gorm.Expr("SELECT id FROM hosts WHERE public_key = ?", publicKey(hk))
```
should work